### PR TITLE
Allow rules that don't start with a positive literal.

### LINKED ISSUE #770: Automatically inject Singleton relation in rules that do not start with a positive literal
Perform the following conversions during parsing:
- `R() :- not R2(...).` into `R() :- Singleton(), not R2(...)`
- `R() :- var x = 5.` into `R() :- Singleton(), var x = 5.`

Currently, users have to perform similar conversion by hand.
This will also address #97.

### PR DESCRIPTION
**Pull request by ryzhyk**
See commit messages.

Resolves #770, #97, and #938.

CC @blp

### CODE REVIEW

File: src/Language/DifferentialDatalog/Validate.hs [hunk 1]
```diff
@@ -303,15 +306,21 @@ indexValidate d idx = do
...
 
 ruleValidate :: (MonadError String me, ?cfg::Config) => DatalogProgram -> Rule -> me Rule
-ruleValidate d rl@Rule{..} = do
-    when (not $ null ruleRHS) $ do
-        case head ruleRHS of
-             RHSLiteral True _ -> return ()
-             _                 -> err d (pos rl) "Rule must start with a positive literal"
-    mapIdxM_ (ruleRHSValidate d rl) ruleRHS
-    mapM_ (ruleLHSValidate d rl) ruleLHS
+ruleValidate d rl = do
```
Comment (mbudiu-vmw) [inline src/Language/DifferentialDatalog/Validate.hs:309]: One would expect validation not to modify the program.
We should have a clearer staging of program rewriting.

**Review by mbudiu-vmw — APPROVED** [review]
I should modify the souffle translator too to get rid of the Dummy relation.

**Comment by mbudiu-vmw** [comment]
It looks like the Windows haskell download still does not work.

**Comment by mbudiu-vmw** [comment]
https://stackoverflow.com/questions/32654493/stack-haskell-throws-tlsexception-in-windows

**Comment by ryzhyk** [comment]
> https://stackoverflow.com/questions/32654493/stack-haskell-throws-tlsexception-in-windows

I saw this, but I don't know what's the command line equivalent of using internet explorer to navigate to a page to fetch the missing certificate.

**Pull request merged**